src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -377,7 +377,7 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                         }
                     }
                     throw new SubscriptionDoesNotBelongToNodeException(
-                        $"Subscription With Id '{_options.SubscriptionName}' cannot be processed by current node, it will be redirected to {appropriateNode}]{Environment.NewLine}Reasons:{string.Join(Environment.NewLine, reasonsDictionary.Select(x => $"{x.Key}:{x.Value}"))}{Environment.NewLine}Message: {connectionStatus.Message}",
+                        $"Subscription With Id '{_options.SubscriptionName}' cannot be processed by current node, it will be redirected to {appropriateNode}]{Environment.NewLine}Reasons:{string.Join(Environment.NewLine, reasonsDictionary.Select(x => $"{x.Key}:{x.Value}"))}{Environment.NewLine}Exception: {connectionStatus.Exception}",
```

Comment:
We should also add the exception as an inner exception, which is better mode all around

Reply:
It is actually a ToString() of the exception
https://github.com/ravendb/ravendb/blob/ba98e423c8737b417fd6e93015f41be68a0de904/src/Raven.Client/Documents/Subscriptions/SubscriptionWorkerOptions.cs#L72

I can create a general `new Exception(message: connectionStatus.Exception)` and set it as a inner exception, but this will request a new ctor of `SubscriptionDoesNotBelongToNodeException`

Reply:
A better option is to have an inner exception, yes. That allow to better track what is going on and simplify the actual error message.